moler/cmd/unix/gunzip.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+"""
+Gunzip command module.
+"""
+
+__author__ = 'Adrianna Pienkowska'
+__copyright__ = 'Copyright (C) 2018, Nokia'
+__email__ = 'adrianna.pienkowska@nokia.com'
+
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Gunzip(GenericUnixCommand):
+    def __init__(self, connection, archive_name, output_file_name=None, options=None, overwrite=False):
+        super(Gunzip, self).__init__(connection=connection)
+        self.archive_name = archive_name
+        self.output_file_name = output_file_name
+        self.options = options
+        self.overwrite = overwrite
+        self.keys = list()
+        self.current_ret['RESULT'] = list()
+
+    def build_command_string(self):
+        cmd = 'gunzip'
+        if self.options:
+            cmd = '{} {}'.format(cmd, self.options)
+        if self.archive_name:
+            for file in self.archive_name:
+                cmd = '{} {}'.format(cmd, file)
+        if self.output_file_name:
+            cmd = '{} > {}'.format(cmd, self.output_file_name)
+        return cmd
+
+    def on_new_line(self, line, is_full_line):
+        if is_full_line:
+            try:
+                self._asks_to_overwrite(line)
+                self._create_dictionary_at_l_option(line)
+                self._command_failure(line)
+                self._parse_line(line)
+            except ParsingDone:
+                pass
+        return super(Gunzip, self).on_new_line(line, is_full_line)
+
+    _re_overwrite = re.compile(r"gzip:\s(?P<FILE_NAME>.*already exists)", re.IGNORECASE)
+
+    def _asks_to_overwrite(self, line):
+        if self._regex_helper.search_compiled(Gunzip._re_overwrite, line):
+            if self.overwrite:
+                self.connection.sendline('y')
```

Comment:
Will send 'y' more times than once, Flag needed - see telnet, ssh, scp for example.

Reply:
Still caught echo from sending 'y' so 'y' will be send twice.

Reply:
Still not corrected. I will make a meeting for F2F review.